core/src/main/java/org/infinispan/interceptors/ClusteringInterceptor.java

```diff
@@ -76,17 +76,19 @@ protected boolean isNeedReliableReturnValues(FlagAffectedCommand command) {
    }
 
    protected boolean needsRemoteGet(InvocationContext ctx, AbstractDataCommand command) {
-      if (command.hasFlag(Flag.CACHE_MODE_LOCAL)
+      if (!command.hasFlag(Flag.DELTA_WRITE)
+         && (command.hasFlag(Flag.CACHE_MODE_LOCAL)
             || command.hasFlag(Flag.SKIP_REMOTE_LOOKUP)
-            || command.hasFlag(Flag.IGNORE_RETURN_VALUES)) {
+            || command.hasFlag(Flag.IGNORE_RETURN_VALUES))) {
          return false;
       }
       boolean shouldFetchFromRemote = false;
       CacheEntry entry = ctx.lookupEntry(command.getKey());
-      if (entry == null || entry.isNull() || entry.isLockPlaceholder()) {
+      if (entry == null || entry.isNull() || entry.isLockPlaceholder()
+         || command.hasFlag(Flag.DELTA_WRITE)) {
          Object key = command.getKey();
          ConsistentHash ch = stateTransferManager.getCacheTopology().getReadConsistentHash();
-         shouldFetchFromRemote = ctx.isOriginLocal() && !ch.isKeyLocalToNode(rpcManager.getAddress(), key) && !dataContainer.containsKey(key);
+         shouldFetchFromRemote = (ctx.isOriginLocal() && !ch.isKeyLocalToNode(rpcManager.getAddress(), key) || command.hasFlag(Flag.DELTA_WRITE)) && !dataContainer.containsKey(key);
```

Comment:
If the key is mapped to the local node in the read CH, there's no need to do a remote fetch. We may need to skip the `ctx.isOriginLocal()` check for `DELTA_WRITE` commands, though.